### ISSUE TITLE
DiagnosticReport Throws System.NullReferenceException in PowerShell Core Beta 9

### ISSUE THREAD

**Issue by pldmgg** [NimaAra/Easy.Common#6]
This is a great little library!

My interest in this is related to PowerShell Core development (PowerShell Core is the .Net 2.0 App that can run on almost any OS, as opposed to Windows PowerShell, which is the .Net Framework App that comes installed on Windows by default). Right now, there are a lot of basic cmdlets missing from PowerShell Core (like Get-NetIPAddress for network info), and this report can help fill-in the gaps until the PowerShell community gets around to creating some of the missing cmdlets.

Since PowerShell Core is a .Net 2.0 App, it should be able to load and use the netstandard2.0 assembly. I can load the assembly without issue, but running Generate() throws the below error:

```
PowerShell v6.0.0-beta.9
Copyright (C) Microsoft Corporation. All rights reserved.

PS C:\Users\testadmin> Add-Type -Path "C:\Users\testadmin\.nuget\packages\Easy.Common.2.6.1\lib\netstandard2.0\Easy.Common.dll"
PS C:\Users\testadmin> [Easy.Common.DiagnosticReport]::Generate()
Unable to generate the Diagnostic Report. Error:
        System.NullReferenceException: Object reference not set to an instance of an object.
   at Easy.Common.DiagnosticReport.<AddProcess>g__Format17_1(String key, Object value, <>c__DisplayClass17_0& )
   at Easy.Common.DiagnosticReport.AddProcess(StringBuilder builder)
   at Easy.Common.DiagnosticReport.GenerateImpl(DiagnosticReportType type)
   at Easy.Common.DiagnosticReport.Generate(DiagnosticReportType type)
```

(As an FYI, the net45 assembly works fine in Windows PowerShell 5.1 - i.e. the .Net Framework App).

(Also, if you haven't worked with PowerShell Core before, you can grab an installer from here: https://github.com/PowerShell/PowerShell/releases)

Thanks again for making this!

**Comment by NimaAra** [comment]
Thanks for reporting this. Here are some questions for you:

1. Is generating the report running on a NON .NET Core app on the same machine successful?
2. Can you try generating the report on the same machine but this time running on a .NET Core app?

Thanks.

**Comment by pldmgg** [comment]
Regarding 1) yes, the report works fine on the non .Net Core App (i.e. .Net Framework App) Windows PowerShell 5.1.

For 2), I can't actually load the .dll in the .Net Core App (i.e. PowerShell Core 6 beta 9), so I can't even attempt to generate a report. The error message when I try to load the netstandard2.0 .dll in PowerShell Core is in my original post. I think the stacktrace is pretty descriptive, but I haven't really done a deep dive into your code.

Let me know if you have any other questions. Thanks again for looking at this!

**Comment by NimaAra** [comment]
At this stage I am not at all interested in _PowerShell_, can you simply create a _.NET Core Console_ application and try generating the report. I just did the same and had no problem generating a report.

The error which you have posted does not relate to _loading_ the dll, the dll has been loaded but the generation of the report seems to have problems; My hunch is _PowerShell Core_ perhaps does not support an _API_ which I am using to generate the report. Hence my request to try this with a working _.NET Core_ app rather than using _PowerShell_.

You should be able to create a _.NET Core Console_ app and reference the _NuGet_ package which will use the _NetStandard2.0_ dll.

**Comment by pldmgg** [comment]
You're right, about the dll loading. I wasn't thinking.

Creating a .Net Core Console App does, indeed, work.

Oh well, maybe I'll dig into `Easy.Common.DiagnosticReport.<AddProcess>` at some point and inspect further.

**Comment by NimaAra** [comment]
Yeah I suspected the problem would be coming from incomplete _API_ in _PowerShell Core_. I am closing this for now as there is not much I can do but feel free to reopen once you have it investigated.

**Comment by NimaAra** [comment]
This should now be fixed in: [v2.6.7](https://github.com/NimaAra/Easy.Common/releases/tag/Easy.Common-release-v2.6.7). Please try again and report back if still a problem.
